internal/cmd/helm-operator/run/cmd.go

```diff
@@ -76,6 +75,12 @@ func NewCmd() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "run",
 		Short: "Run the operator",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if cmd.Flag("metrics-require-rbac").Changed && !cmd.Flag("metrics-secure").Changed {
+				return fmt.Errorf("--metrics-secure flag is required when --metrics-require-rbac is present")
```

Comment:
1. `Errorf` is used, but no formatting directive is provided. Swap it for `errors.New`?
2. `*pflag.Flag{}.Changed` isn't doing the right comparison. You want to check the _value_, not just that it's changed. If someone (for whatever reason) calls `--metrics-require-rbac=false`, this logic would throw an error because `cmd.Flag("metrics-require-rbac").Changed` is true. There are other failure cases as well, similar to this one.